lib/datasets/downloader.rb

```diff
@@ -31,41 +33,59 @@ def download(output_path)
         headers["Range"] = "bytes=#{start}-"
       end
 
-      Net::HTTP.start(@url.hostname,
-                      @url.port,
-                      :use_ssl => (@url.scheme == "https")) do |http|
-        path = @url.path
-        path += "?#{@url.query}" if @url.query
-        request = Net::HTTP::Get.new(path, headers)
+      start_http(@url, headers) do |response|
+        case response
+        when Net::HTTPPartialContent
+          mode = "ab"
+        when Net::HTTPSuccess
+          start = nil
+          mode = "wb"
+        else
+          break
+        end
```

Comment:
We can assume that `response` is a success response here. So how about simplify this like the following?

```suggestion
        if response.is_a?(Net::HTTPPartialContent)
          mode = "ab"
        else
          start = nil
          mode = "wb"
        end
```